index.js

```diff
@@ -186,39 +127,51 @@ function makeCachedResolver (prefix, fieldName, cache, originalFieldResolver, po
       }
 
       // We must skip ctx and info as they are not easy to serialize
+      // TODO use a fast JSON stringify
+      // TODO can we remove self from the key?
       return { self, arg, fields, extendKey }
     }
-  }, async function ({ self, arg, ctx, info, extendKey }, key) {
-    if (storage) {
-      const val = await storage.get(name + '~' + key)
-      if (val) {
-        onHit()
-        return val
-      }
-    }
-
-    onCacheMiss ? onCacheMiss() : onMiss()
-    const res = await originalFieldResolver(self, arg, ctx, info)
-    if (storage) {
-      await storage.set(name + '~' + key, res)
-    }
-    return res
+  }, async function ({ self, arg, ctx, info }) {
+    return originalFieldResolver(self, arg, ctx, info)
   })
+
   return async function (self, arg, ctx, info) {
-    const isMutation =
-      info.operation && info.operation.operation === 'mutation'
-    if (
-      (skip && (await skip(self, arg, ctx, info))) ||
-      isMutation ||
-      (policy && policy.skip && (await policy.skip(self, arg, ctx, info)))
-    ) {
-      if (!isMutation) {
-        onCacheSkip ? onCacheSkip() : onSkip()
+    let result
+    try {
+      // dont use cache on mutation
+      // TODO dont cache also subscriptions
+      if (info.operation && info.operation.operation === 'mutation') {
+        result = await originalFieldResolver(self, arg, ctx, info)
+      } else if (
+        (skip && (await skip(self, arg, ctx, info))) ||
+        (policy && policy.skip && (await policy.skip(self, arg, ctx, info)))
+      ) {
+        // dont use cache on skip by policy or by general skip
+        report[name].onSkip()
+        result = await originalFieldResolver(self, arg, ctx, info)
+      } else {
+        // use cache to get the result
+        result = await cache[name]({ self, arg, ctx, info })
       }
+
+      if (invalidate) {
+        // note: invalidate is async but no await
+        invalidation(invalidate, cache, name, self, arg, ctx, info, result)
+      }
+    } catch (err) {
+      // TODO implement onError
```

Comment:
what are we doing with the error?

Reply:
with the current version, nothing
happy to implement `onError`

Reply:
Yep, add that ;).